editing/edit-with-branch-versioning/src/main/java/com/esri/samples/edit_with_branch_versioning/EditWithBranchVersioningController.java

```diff
@@ -0,0 +1,359 @@
+                mapView.setViewpointAsync(new Viewpoint(featureLayer.getFullExtent()));
+                progressIndicator.setVisible(false);
+                createVersionButton.setDisable(false);
+                currentVersionLabel.setText("Current version: " + serviceGeodatabase.getVersionName());
+              } else {
+                new Alert(Alert.AlertType.ERROR, "Feature layer failed to load").show();
+              }
+            });
+          } else {
+            new Alert(Alert.AlertType.ERROR, "Unable to get the service feature table").show();
+          }
+        } else {
+          progressIndicator.setVisible(false);
+          new Alert(Alert.AlertType.ERROR, "Service geodatabase failed to load").show();
+        }
+      });
+
+      // listen to clicks on the map to select or move features
+      mapView.setOnMouseClicked(event -> {
+        if (event.isStillSincePress() && event.getButton() == MouseButton.PRIMARY) {
+          // reset the UI
+          featureLayer.clearSelection();
+          editFeatureVBox.setDisable(true);
+
+          // select the clicked feature
+          Point2D point = new Point2D(event.getX(), event.getY());
+          selectFeature(point);
+        }
+
+        if (event.isStillSincePress() && event.getButton() == MouseButton.SECONDARY) {
+          // if a feature is selected and the current version is not the default, update the feature's geometry
+          if (selectedFeature != null && !serviceGeodatabase.getVersionName().equals(defaultVersion)) {
+            Point2D point = new Point2D(event.getX(), event.getY());
+            Point mapPoint = mapView.screenToLocation(point);
+            selectedFeature.setGeometry(mapPoint);
+            updateFeature(selectedFeature);
+          }
+        }
+      });
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Create a new branch version using user defined values.
+   */
+  @FXML
+  private void handleCreateVersionButtonClicked() {
+
+    // validate version name input
+    String inputName = nameTextField.getText();
+    if (inputName.contains(".") || inputName.contains(";") || inputName.contains("'") || inputName.contains("\"")) {
+      new Alert(Alert.AlertType.ERROR,
+        "Please enter a valid version name.\nThe name cannot contain the following characters:\n. ; ' \" ").show();
+      return;
+    } else if (inputName.length() > 0 && Character.isWhitespace(nameTextField.getText().charAt(0))) {
+      new Alert(Alert.AlertType.ERROR, "Version name cannot begin with a space").show();
+      return;
+    } else if (inputName.length() > 62) {
+      new Alert(Alert.AlertType.ERROR, "Version name must not exceed 62 characters").show();
+      return;
+    } else if (inputName.length() == 0) {
+      new Alert(Alert.AlertType.ERROR, "Please enter a version name").show();
+      return;
+    }
```

Comment:
I'd be tempted to abstract this out into its own method (`validateTextInput`). This is quite a chunk of code for handling input which detracts from the main point of this method, which is to show the `ServiceVersionParameters` etc functionality. We don't often go for individual methods but in a case like this where there is a lot of UI specific case handling, it's ok to move that out.